bazel/envoy_build_system.bzl

```diff
@@ -20,12 +20,45 @@ def envoy_copts(repository, test = False):
         repository + "//bazel:dbg_build": ["-ggdb3"],
     }) + select({
         repository + "//bazel:disable_tcmalloc": [],
+        "//bazel/osx:darwin": [],
```

Comment:
Is this the permanent plan? Does gperftools support OS X? tcmalloc has some performance advantages in addition to performance insight via the tools.